tietokanta/src/main/resources/db/migration/V1_839__.sql

```diff
@@ -0,0 +1,127 @@
+UPDATE toteuma_200701_210101 set poistettu = false WHERE poistettu is null;
+UPDATE toteuma_210101_210701 set poistettu = false WHERE poistettu is null;
+UPDATE toteuma_210701_220101 set poistettu = false WHERE poistettu is null;
+UPDATE toteuma_220101_220701 set poistettu = false WHERE poistettu is null;
+UPDATE toteuma_220701_230101 set poistettu = false WHERE poistettu is null;
+UPDATE toteuma_230101_230701 set poistettu = false WHERE poistettu is null;
+UPDATE toteuma_230701_240101 set poistettu = false WHERE poistettu is null;
+UPDATE toteuma_240101_240701 set poistettu = false WHERE poistettu is null;
+UPDATE toteuma_240701_250101 set poistettu = false WHERE poistettu is null;
+UPDATE toteuma_250101_991231 set poistettu = false WHERE poistettu is null;
+-- Sitten muokataan itse kenttä, jotta sinne ei voi null arvoja enää lisätä
+ALTER TABLE toteuma ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_010101_191001 ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_191001_200701 ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_200701_210101 ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_210101_210701 ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_210701_220101 ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_220101_220701 ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_220701_230101 ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_230101_230701 ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_230701_240101 ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_240101_240701 ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_240701_250101 ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_250101_991231 ALTER COLUMN poistettu SET NOT NULL;
+-- set default false
+ALTER TABLE toteuma ALTER COLUMN poistettu SET DEFAULT FALSE;
+ALTER TABLE toteuma_010101_191001 ALTER COLUMN poistettu SET DEFAULT FALSE;
+ALTER TABLE toteuma_191001_200701 ALTER COLUMN poistettu SET DEFAULT FALSE;
+ALTER TABLE toteuma_200701_210101 ALTER COLUMN poistettu SET DEFAULT FALSE;
+ALTER TABLE toteuma_210101_210701 ALTER COLUMN poistettu SET DEFAULT FALSE;
+ALTER TABLE toteuma_210701_220101 ALTER COLUMN poistettu SET DEFAULT FALSE;
+ALTER TABLE toteuma_220101_220701 ALTER COLUMN poistettu SET DEFAULT FALSE;
+ALTER TABLE toteuma_220701_230101 ALTER COLUMN poistettu SET DEFAULT FALSE;
+ALTER TABLE toteuma_230101_230701 ALTER COLUMN poistettu SET DEFAULT FALSE;
+ALTER TABLE toteuma_230701_240101 ALTER COLUMN poistettu SET DEFAULT FALSE;
+ALTER TABLE toteuma_240101_240701 ALTER COLUMN poistettu SET DEFAULT FALSE;
+ALTER TABLE toteuma_240701_250101 ALTER COLUMN poistettu SET DEFAULT FALSE;
+ALTER TABLE toteuma_250101_991231 ALTER COLUMN poistettu SET DEFAULT FALSE;
+
+
+
+-- Muokataan toteuma_tehtava taulu suorituskyvyltään paremmaksi
+-- Lisätään urakkaindeksiä varten uusi columni
+ALTER TABLE toteuma_tehtava ADD COLUMN urakka_id INTEGER;
```

Comment:
Tämähän luo redundanssia, kun urakka-id on toteumaviitteen kautta johdettavissa. Kenties kuitenkin ihan OK, jos on todennettavissa suorituskykyhyöty.

Reply:
Pitäskö tuohon tehdä samalla foreign key constraint  `urakka` taulukkoon?

Reply:
Erittäin hyvä nosto. Jos tuollainen haluttaisiin, niin se pitäisi olla urakka<--->toteuma taulujen välillä (mitä ei ole) ja sitten tähän tauluun se olisi pitänyt laittaa toteuma <---> toteuma_tehtava linkiksi. (jota ei ole) Tehdäänkö tästä erillinen bugi?